### DISCUSSION TITLE
regex in file-based routing

### DISCUSSION THREAD

**Discussion by ArianHamdi** [vercel/next.js#26046] [Help]
I want to show a SSR page for /product/12 and a SSG page for /product/Mobile-Samsung
I know I can use different URL to do this, but how to define a page for /product/number and /product/string 

I think it would be nice to use regex in the file name to determine which page should be fired.

**Comment by akellbl4** [comment]
Hi @ArianHamdi,

I think you can render different views for different URL params.
I could suggest something like this:

```js
// pages/product/[id].js

export async function getServerSideProps/getStaticProps(ctx) {
  if (/^\d+$/.test(ctx.params.id)) {
     const data = await fetchById(ctx.params.id)

    return {
      props: {data, param:  'id'}
    }
  }

  const data = await fetchBySlug(ctx.params.id)

  return {
    props: { data, param: 'slug' }
  }
}


export default function Page({ param, data }) {

  if (param === 'slug') {
    return <ViewBySlug data={data} />
  }

  return <ViewBySlug data={data} />
}
```

**Reply by ArianHamdi** [reply]
Hi Pavel
Thank you for the attention.
I know I can render different view for different URL params but in this way I should use one type of pre-rendering for two views.
so, what can I do if I want SSR for product/number and SSG for product/string ?

**Reply by akellbl4** [reply]
Got it. Interesting question. As far as I know, it's no way to do it with the current Next.js API.
Maybe you can create different pages and handle them with redirects, but it doesn't look like a good solution for your problem at all.

I see it like:
```js
export async function getServerSideProps/getStaticProps(ctx) {
  if (/^\d+$/.test(ctx.params.id)) {
     const data = await fetchById(ctx.params.id)

    return {
      props: {data, param:  'id'}
    }
  }

  return {
    redirect: { ... }
  }
}
```
